contracts/strategies/allocation/AllocationWithVotingBase.sol

```diff
@@ -226,28 +235,41 @@ contract AllocationWithVotingBase is IAllocationStrategy, Ownable, Initializable
         }
 
         // Decode the _data
-        Allocations[] memory allocationsPerToken = abi.decode(_data, Allocations[]);
+        (Allocations[] memory allocationsPerToken) = abi.decode(_data, (Allocations[]));
         for (uint256 i; i < allocationsPerToken.length;) {
             // Get the allocations per token
             Allocations memory allocations = allocationsPerToken[i];
 
             uint256 allocationsTotalAmount = allocations.totalAmount;
 
+            // convert to TransferData
+            Transfer.TransferData memory transferData =
+                Transfer.TransferData({to: address(this), amount: allocationsTotalAmount});
+
             // transfer total amount of token to this contract
-            Transfer.transferAmount(allocations.token, allocationsTotalAmount);
+            Transfer.transferAmount(allocations.token, transferData);
 
-            for (uint256 i = 0; i < allocations.data.length;) {
+            for (uint256 j = 0; j < allocations.data.length;) {
                 if (allocationsTotalAmount == 0) {
                     revert ALLOCATION_AMOUNT_UNDERFLOW();
                 }
 
-                Application storage application = applications[allocations.data.applicationId];
-                application.tokenAmounts[allocations.token] += allocations.data.amount;
+                Application storage application = applications[allocations.data[j].applicationId];
+                application.tokenAmounts[allocations.token] += allocations.data[j].amount;
 
-                allocationsTotalAmount -= allocations.data.amount;
+                allocationsTotalAmount -= allocations.data[j].amount;
+
+                // TOOD: check
+                emit ApplicationVoted(
+                    sender,
+                    allocations.data[j].applicationId,
+                    allocations.data[j].amount,
+                    allocations.token,
+                    application.recipient
+                );
```

Comment:
^ we don't want to emit a vote for every vote. This gets quite expensive 
 Let's revert to the older format.

Reply:
I moved it outside the `for` loop and added a comment. Also added a new event for multiple at the same time. We can either keep both or eliminate the singular event.

` emit ApplicationsVoted(sender, new uint256[](0), new uint256[](0), new address[](0));`

This still needs to be finished. I am not sure yet how to derive the data to pass to the event.